_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+- last release from `master` is `1.1.0`
+- last release from `next` is `2.0.0`
+
+Committing a breaking change to `master` would trigger the release of `2.0.0` that would fail as the release already exists.
+
+In addition when the `master` and `next` have the same commit head (after a merging `next` into `master` for example), the first commit done on `next` must be a breaking change.
+Otherwise that would result in a "fork" situation.
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `2.0.0`
+- last release from `next` is `2.0.0`
+
+A `feat` commit on `master` will release version `2.1.0`.
+Committing a `feat` commit on `next` would trigger the release of `2.1.0` that would fail as the release already exists.
+
+With specific channel names:
+
+```json
+{
+  "release": {
+    "branches": [
+      "master",
+      {"branch": "next", "channel": "preview"}
+    ]
+  }
+}
+```
+
+#### LTS branches
+
+Workflow to release legacy versions.
+
+Ideal for modules maintaining legacy versions and doing bug fixes and features only releases.
+
+Branch: `1.x`, `2.x`, `master` (any number of lts branches are supported)
+
+Channel: `1.x`, `2.x`, `latest` if supported by the release target
+
+##### LTS branches workflow
+
+| Action                                          | Result                                                                                                                  |
+|-------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| **push** to `master`                            | Release on default channel (if supported, otherwise no channel)                                                         |
+| **merge** feature branch to `master`            | Release on default channel (if supported, otherwise no channel)                                                         |
+| **push** to `1.x` or `2.x` branches             | Release on `1.x` or `2.x` channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+| **merge** any branch to `1.x` or `2.x` branches | Release on `1.x` or `2.x` channel (if supported, otherwise no channel) if in range, report an error and error otherwise |
+
+##### LTS branches configuration
+
+```json
+{
+  "release": {
+    "branches": ["1.x", "2.x", "master"]
+  }
+}
+```
+
+With specific channel names:
+
+```json
+{
+  "release": {
+    "branches": [
+      {"branch": "v1", "range": "1.x", "channel": "v1"},
+      {"branch": "v2", "range": "2.x", "channel": "v2"},
+      "master"
+    ]
+  }
+}
+```
+
+#### Pre-releases
+
+Workflow to distribute releases without incrementing the semantic version during the development of new a application version.  
+Ideal for applications distributing unstable/alpha/preview releases.
+
+Branch: `master`, `beta`, `alpha`
```

Comment:
say I’m on version 2.3.4. Then I push a breaking change to `beta`, which will release `3.0.0-beta.1`. Right?

Now time moves on and I'm at version 4.5.6. I want to publish another beta release. But the beta branch is super outdated. Wouldn't that cause problems? Would I need to force push to the branch?

How about we set the convention for pre-release branches to contain a base version number. So in my example above, I would push a branch `3.0.0-beta` and later I would push a branch `5.0.0-beta`. That would avoid conflicts. And `"beta"` could be anything, so multiple people/teams could spike out features for the next major version at the same time, using different pre-release names

Reply:
> Now time moves on and I'm at version 4.5.6. I want to publish another beta release. But the beta branch is super outdated. Wouldn't that cause problems? Would I need to force push to the branch?

That would be handled by Git. If you push to you outdated branch you are going to continue to release within this release line `3.0.0-beta.X`. I think that's the expected behavior as the release number has to represent what's in it, in that case the code of the `3.0.0-beta` plus some new code.

If you want to release a new beta based on the `4.5.6` release line, then you have to bring all the code included in `4.5.6` into your beta branch and add you new code on top of it. That would release a `5.0.0-beta` (assuming you push a breaking change on beta)

Reply:
> If you want to release a new beta based on the 4.5.6 release line, then you have to bring all the code included in 4.5.6 into your beta branch and add you new code on top of it. That would release a 5.0.0-beta (assuming you push a breaking change on beta)

That would be a really neat feature!

On a related note, how would I release a beta feature on `5.0.0`, while a team member releases their own beta feature on `5.0.0` (where neither of our features depend on one another, we're simply working independently on new features)